--- stat-genconfig	2020-01-12 21:15:03.000000000 -0500
+++ stat-genconfig	2020-01-12 21:15:01.000000000 -0500
@@ -176,6 +176,54 @@
 		   .. _bool(   c["Forward"],    "Forward"    )
 end
 
+function config_mqtt( c )
+  local str = ""
+
+  for s in pairs(sections) do
+    for key, type in pairs({ Sub="collectd_mqtt_sub", Pub="collectd_mqtt_pub" }) do
+      if sections[s][".type"] == type then
+        sname = sections[s].sname
+        host = sections[s].host
+        user = sections[s].user
+        port = sections[s].port
+        topic = sections[s].topic
+        password = sections[s].password
+        pname = sections[s].pname
+        prefix = sections[s].prefix
+
+        if sname then
+          if user then
+            str = str .. "\t" .. "<Subscribe " .."\"".. sname .."\"".." >" .. " \n"
+            str = str .. "\t" .. "host" .. " \"" .. host .. "\"\n"
+            str = str .. "\t" .. "port" .. " \"" .. port .. "\"\n"
+            str = str .. "\t" .. "user" .. " " .. user .. "\n"
+            str = str .. "\t" .. "password" .. " " .. password .. "\n"
+            str = str .. "\t" .. "topic" .. " \"" .. topic .. "\"\n"
+            str = str .. "\t" .. "CleanSession true" .. " \n"
+            str = str .. "\t" .. "</Subscribe>" .. " \n"
+          else
+            str = str .. "\t" .. "<Subscribe " .."\"".. sname .."\"".." >" .. " \n"
+            str = str .. "\t" .. "host" .. " \"" .. host .. "\"\n"
+            str = str .. "\t" .. "port" .. " \"" .. port .. "\"\n"
+            str = str .. "\t" .. "topic" .. " \"" .. topic .. "\"\n"
+            str = str .. "\t" .. "</Subscribe>" .. " \n"
+          end
+        else
+          str = str .. "\t" .. "<Publish " .."\"".. pname .."\"".." >" .. " \n"
+          str = str .. "\t" .. "Host" .. " \"" .. host .. "\"\n"
+          if port ~= nil then
+            str = str .. "\t" .. "Port" .. " \"" .. port .. "\"\n"
+          end
+          str = str .. "\t" .. "Prefix" .. " \"" .. prefix .. "\"\n"
+          str = str .. "\t" .. "Retain true\n"
+          str = str .. "\t" .. "</Publish>" .. " \n"
+        end
+      end
+    end
+  end
+
+  return str
+end
 
 function _list_expand( c, l, nopad )
 	local str = ""
@@ -452,6 +500,8 @@
 		{ },
 		{ }
 	},
+
+  mqtt = config_mqtt,
 }
 
 preprocess = {
