--- luci_statistics.lua.bkp	2020-01-12 21:39:34.000000000 -0500
+++ luci_statistics.lua	2020-01-12 21:43:28.000000000 -0500
@@ -54,12 +54,13 @@
 		tcpconns	= _("TCP Connections"),
 		thermal	= 	_("Thermal"),
 		unixsock	= _("UnixSock"),
-		uptime		= _("Uptime")
+		uptime		= _("Uptime"),
+    mqtt      = _("MQTT")
 	}
 
 	-- our collectd menu
 	local collectd_menu = {
-		output  = { "csv", "network", "rrdtool", "unixsock" },
+		output  = { "csv", "network", "rrdtool", "unixsock", "mqtt" },
 		general = { "apcups", "contextswitch", "cpu", "cpufreq", "df",
 			"disk", "email", "entropy", "exec", "irq", "load", "memory",
 			"nut", "processes", "sensors", "thermal", "uptime" },
